soroban-env-host/src/native_contract/token/balance.rs

```diff
@@ -443,7 +443,11 @@ fn transfer_account_balance(e: &Host, account_id: AccountId, amount: i64) -> Res
         };
         if new_balance >= min_balance && new_balance <= max_balance {
             ae.balance = new_balance;
-            le = Host::ledger_entry_from_data(e, LedgerEntryData::Account(ae))?;
+            le = Host::ledger_entry_from_data(
```

Comment:
Great catch! This still seems a bit error-prone though. Should we split this into separate methods? 

- Create a new contract data/code entry accepting the respective `ContractCode/DataEntry` struct (host doesn't create any other LE types). V0 ext by default would be fine there
- Modify data of a provided LE. This would ensure the ext is never missed when we just modify an entry.

WDYT?

Reply:
To be able to modify the LE, I think I'd have to clone it after calling `storage.get`. Are there any concerns with that?

Reply:
You can leave implementation the same, just update the method interface for clarity, such as

```
fn modify_ledger_entry_data(host, original_entry, new_data) {
  LedgerEntry {
                last_modified_ledger_seq: 0,
                data: new_data,
                ext: original_entry.ext.metered_clone(),
   }
```

Basically the same implementation as we have now, just less room for error (as there is no way to accidentally pass the default ext).

Reply:
Ah I see what you're saying. Updated.